pkg/occlient/occlient.go

```diff
@@ -2006,67 +2009,103 @@ func (c *Client) DeleteServiceInstance(labels map[string]string) error {
+//
+// "Projects are deleted asynchronously after you send the delete command. So it's possible that the deletion just hasn't been reconciled yet. It should happen within a minute or so, so try again.
+// Also, please be aware that in a multitenant environment, like OpenShift Online, you are prevented from creating a project with the same name as any other project in the cluster, even if it's not your own. So if you can't create the project, it's possible that someone has already created a project with the same name."
+func (c *Client) DeleteProject(name string, wait bool) error {
+
+	// Instantiate watcher for our "wait" function
+	var watcher watch.Interface
+	var err error
+
+	// If --wait has been passed, we will wait for the project to fully be deleted
+	if wait {
+		watcher, err = c.projectClient.Projects().Watch(metav1.ListOptions{
+			FieldSelector: fields.Set{"metadata.name": name}.AsSelector().String(),
+		})
+		if err != nil {
+			return errors.Wrapf(err, "unable to watch project")
+		}
+		defer watcher.Stop()
 	}
 
-	// wait for delete to complete
-	w, err := c.projectClient.Projects().Watch(metav1.ListOptions{
-		FieldSelector: fields.Set{"metadata.name": name}.AsSelector().String(),
-	})
+	// Delete the project
+	err = c.projectClient.Projects().Delete(name, &metav1.DeleteOptions{})
 	if err != nil {
-		return errors.Wrapf(err, "unable to watch project")
+		return errors.Wrap(err, "unable to delete project")
 	}
 
-	defer w.Stop()
-	for {
-		val, ok := <-w.ResultChan()
-		// When marked for deletion... val looks like:
-		/*
-			val: {
-				Type:MODIFIED
-				Object:&Project{
-					ObjectMeta:k8s_io_apimachinery_pkg_apis_meta_v1.ObjectMeta{...},
-					Spec:ProjectSpec{...},
-					Status:ProjectStatus{
-						Phase:Terminating,
-					},
-				}
-			}
-		*/
-		// Post deletion val will look like:
-		/*
-			val: {
-				Type:DELETED
-				Object:&Project{
-					ObjectMeta:k8s_io_apimachinery_pkg_apis_meta_v1.ObjectMeta{...},
-					Spec:ProjectSpec{...},
-					Status:ProjectStatus{
-						Phase:,
-					},
-				}
-			}
-		*/
-		if !ok {
-			return fmt.Errorf("received unexpected signal %+v on project watch channel", val)
-		}
-		// So we depend on val.Type as val.Object.Status.Phase is just empty string and not a mapped value constant
-		if prj, ok := val.Object.(*projectv1.Project); ok {
-			glog.V(4).Infof("Status of delete of project %s is %s", name, prj.Status.Phase)
-			switch prj.Status.Phase {
-			//prj.Status.Phase can only be "Terminating" or "Active" or ""
-			case "":
-				if val.Type == watch.Deleted {
-					return nil
+	// If watcher has been created (wait was passed) we will create a go routine and actually **wait**
+	// until *EVERYTHING* is successfully deleted.
+	if watcher != nil {
+
+		// Project channel
+		// Watch error channel
+		projectChannel := make(chan *projectv1.Project)
+		watchErrorChannel := make(chan error)
+
+		// Create a go routine to run in the background
+		go func() {
+
+		loop:
```

Comment:
Why are we using a label here? Since it's a single loop, why not break normally?

Reply:
Removing this :+1: you're right.